src/shared/sam/cli/samCliLocator.ts

```diff
@@ -29,9 +29,9 @@ export class DefaultSamCliLocationProvider implements SamCliLocationProvider {
      * Gets the last-found `sam` location, or searches the system if a working
      * `sam` wasn't previously found and cached.
      */
-    public async getLocation() {
+    public async getLocation(forceSearch?: boolean) {
         const perflog = new PerfLog('samCliLocator: getLocation')
-        const cachedLoc = DefaultSamCliLocationProvider.cachedSamLocation
+        const cachedLoc = forceSearch ? undefined : DefaultSamCliLocationProvider.cachedSamLocation
 
         // Avoid searching the system for `sam` (especially slow on Windows).
         if (cachedLoc && (await DefaultSamCliLocationProvider.isValidSamLocation(cachedLoc.path))) {
```

Comment:
Probably will remove `isValidSamLocation` in a future PR. In the worst case (somehow the cached `sam` breaks), the user must either restart vscode or run "Detect SAM CLI".